modules/041-linstor/templates/linstor-csi/linstorcsidriver.yaml

```diff
@@ -170,9 +170,23 @@ spec:
       maxUnavailable: 1
     {{- end }}
   controllerEndpoint: https://linstor.d8-{{ .Chart.Name }}.svc:3371
+  {{- if and (.Values.linstor.dataNodes) (.Values.linstor.dataNodes.nodeGroups) }}
+  nodeAffinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - operator: In
+                key: node.deckhouse.io/group
+                values:
+                {{- range $nodeGroup := .Values.linstor.dataNodes.nodeGroups }}
+                - {{ $nodeGroup | quote }}
+                {{- end}}
+  {{- else }}
   nodeAffinity: {}
+  {{- end }}
   nodeTolerations:
-    {{- index (include "helm_lib_tolerations" (tuple . "wildcard") | fromYaml) "tolerations" | toYaml | nindent 4 }}
+    {{- index (include "helm_lib_tolerations" (tuple . "any-node") | fromYaml) "tolerations" | toYaml | nindent 4 }}
```

Comment:
Same is here, if csi-node pod will be evicted before the workload, the volumes will stay unmanageable for this node.